geoAssembler/optimiser/centre.py

```diff
@@ -0,0 +1,109 @@
+from collections import namedtuple
+from typing import Union, Tuple
+
+import numpy as np
+from extra_geom.detectors import DetectorGeometryBase
+from scipy.optimize import differential_evolution
+
+from .utility import Integrator
+
+
+class CentreOptimiser:
+    """
+    Object used to manage the optimisation of the centre position
+    via maximization of the azimuthal integration result.
+
+    The centre position leading to the highest azimuthal integration
+    value at a radial bin should be the one where the diffraction
+    rings form as straight a line as possible, and the rings becoming
+    straight lines in polar coordinates indicates an accurate centre.
+    """
+    def __init__(self, geom: DetectorGeometryBase,
+                 stacked_mean: np.ndarray, sample_dist_m: Union[int, float],
+                 unit: str = "2th_deg"):
```

Comment:
Add maybe a docstring that describes what are the input arguments.

Reply:
Also `stack_mean` is maybe misleading, as I suppose it does not have to be a mean image to be usable?